src/Entities/Assertions/AssertionFactory.php

```diff
@@ -314,54 +320,104 @@ public function getInstance(\stdClass $annotation)
             case Invariant::ANNOTATION:
             case Requires::ANNOTATION:
                 // complex annotations leave us with two possibilities: raw or custom assertions
-
                 if (isset($annotation->values['type'], $annotation->values['constraint'])) {
                     // we need a custom assertion here
-
-                    $potentialAssertion = $annotation->values['type'];
-                    if (class_exists($potentialAssertion)) {
-                        $assertionInstance = new $potentialAssertion($annotation->values['constraint']);
-                        if (!$assertionInstance instanceof AssertionInterface) {
-                            throw new \Exception(
-                                sprintf(
-                                    'Specified assertion of type %s does not implement %s',
-                                    $potentialAssertion,
-                                    AssertionInterface::class
-                                )
-                            );
-                        }
-                        return $assertionInstance;
-                    }
-
-                    $potentialAssertion = '\AppserverIo\Doppelgaenger\Entities\Assertions\\' . $annotation->values['type'] . 'Assertion';
-                    if (class_exists($potentialAssertion)) {
-                        // we know the class! Create an instance using the passed constraint
-                        /** @var \AppserverIo\Psr\MetaobjectProtocol\Dbc\Assertions\AssertionInterface $assertionInstance */
-                        $assertionInstance = new $potentialAssertion($annotation->values['constraint']);
-                        return $assertionInstance;
-
-                    } else {
-                        throw new \Exception(sprintf('Cannot create complex assertion of type %s', $annotation->values['type']));
-                    }
-
-                } else {
-                    // a RawAssertion is sufficient
-
-                    return new RawAssertion(array_pop($annotation->values));
+                    return $this->createAssertion($annotation->values['type'], $annotation->values['constraint']);
                 }
-                break;
```

Comment:
I would leave the `break` statement there. It does not hurt but if someone might remove the `return` in the future the missing `break` statement will hurt indeed.